### DISCUSSION TITLE
How to style a column of data table.

### DISCUSSION THREAD

**Discussion by listeng** [Budibase/budibase#3923] [Help] [answered]
Hi all. I want to make a custom style for a column in data table by the field value. For a example, the row in table, if the value of the Status field is 1 then show color green, if the value is 0 then show color red. I can not find any document for it, How to do it? Thanks.

**Comment by mjashanks** [comment, marked as answer]
It's not possible at the moment, though we do have a long list of table improvements that we plan to do in the next few months: https://github.com/Budibase/budibase/discussions/1913

I have added your comment in there - thanks!

**Comment by Toucan14** [comment]
Hey all, any news on that? Would be really helpful right now. Thanks!

**Comment by christophsanz** [comment]
I found a workaround:

1. Add a custom color to the column you want to add css to and give it some name (does not have to be a valid color)
![image](https://user-images.githubusercontent.com/22081130/211563882-d1e1ead5-759d-4527-996c-9751145e07c0.png)
2. add an embed with the following code, referencing your "color" :

```
<style>
.spectrum-Table-cell[style="color: whatever;"]
{
     font-weight: bold; // add whatever css your heart desires
}
</style>
```

The same should also work with the background color to style the entire cell, but i haven't tried this.

**Reply by rafikApp** [reply]
hey @christophsanz
I tried but it didn't work. any help please?

**Reply by christophsanz** [reply]
What exactly does not work?
You basically just use the css attribute selector in combination with the style tag.
Try it only with the text color if you have also a background color applied.
You can add the background color in your css again.

**Reply by rafikApp** [reply]
Hey @christophsanz, thank you for your help.
I gave the name (test) to the text color of the column

![t1](https://user-images.githubusercontent.com/106697208/212161746-264bb8e9-7704-4682-9b6f-3c70cbe46ef9.PNG)

 and added an embed to the table which contains the following code.
 ```
{{#eq New Table.CONSIGNATIONS.statut "A confirmée" }}
<style>
.spectrum-Table-cell[style="color:test;"]
{
     font-weight: bold; 
      color: green
}
</style>
{{ else }}
<style>
.spectrum-Table-cell[style="color:test;"]
{
     font-weight: bold; 
     color: black
}
{{/eq}}
```

![t2](https://user-images.githubusercontent.com/106697208/212162033-58f44ee2-fb35-45e9-8b34-21c15df02e9a.PNG)


but the status is always displayed in green regardless of its value

**Reply by christophsanz** [reply]
I think the space between "color:" and "test"  is important.
Also i am not sure if you need a semicolon on color: black;
Also try with "!important" and check the chrome devt-tools for colliding styles or if the styles attribute is written exactly like you do.

**Reply by rafikApp** [reply]
Yes i tried, but the same probleme

![Capture](https://user-images.githubusercontent.com/106697208/212172202-4632a4c4-522d-46a3-ad3a-269efe115e81.PNG)

**Comment by christophsanz** [comment]
Looks like the comparison with #eq is your problem.
try #is instead of #eq

**Reply by rafikApp** [reply]
No i don't think.
It works for you ?

**Comment by NicolasBio** [comment]
Hi,
I was wondering if this still works using the current table component? 
I'm looking for column setting to color a column but can't seem to find it.

**Comment by mikesealey** [comment]
@listeng @christophsanz @rafikApp @Toucan14 

We're looking into adding extra styling controls to tables and would love to hear your thoughts. Please head to Discussion https://github.com/Budibase/budibase/discussions/13918#discussioncomment-9772139 and leave a reply. The better we can understand our communities needs the better we can serve them.
